modules/javafx.graphics/src/main/java/com/sun/javafx/text/PrismLayoutInfo.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.sun.javafx.text;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javafx.geometry.Insets;
+import javafx.geometry.Rectangle2D;
+import javafx.scene.text.CaretInfo;
+import javafx.scene.text.LayoutInfo;
+import javafx.scene.text.TextLineInfo;
+import com.sun.javafx.geom.BaseBounds;
+import com.sun.javafx.scene.text.TextLayout;
+import com.sun.javafx.scene.text.TextLine;
+
+/**
+ * Layout information as reported by PrismLayout.
+ */
+public abstract class PrismLayoutInfo extends LayoutInfo {
+
+    protected abstract double lineSpacing();
+
+    protected abstract Insets insets();
+
+    private final TextLayout layout;
+
+    public PrismLayoutInfo(TextLayout layout) {
+        this.layout = layout;
+    }
+
+    @Override
+    public Rectangle2D getBounds(boolean includeLineSpacing) {
+        BaseBounds b = layout.getBounds();
+        Insets m = insets();
```

Comment:
See my comments below about adding the insets to the layout information. Maybe we could also add another boolean `includeInsets` for this?

Reply:
Good question!  I don't think this is needed: the call returns the layout bounds (in the owning node coordinate system).

The insets only change the position of the layout within the owning node, unlike lineSpacing which actually changes the shape by adding or removing the last line spacing.

I think this method is ok as is.